internal/document_sync/document_sync.go

```diff
@@ -0,0 +1,30 @@
+package document_sync
```

Comment:
I think we should move this file into the internal/lsp directory and add it to the lsp package